### PR TITLE
[RFR] Adding tests for appliance_console_cli

### PR DESCRIPTION
**Pull request by lcouzens**
# Purpose or Intent
- Adding appliance_console_cli class 
- Adding test to appliance_console_cli

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit b3273504c24e91966283380b5a1533fb5f865251:

`utils/appliance/__init__.py`:
- :bangbang: Line 53:1: **E302** *expected 2 blank lines, found 1*
- :bangbang: Line 59:17: **E231** *missing whitespace after ','*
- :bangbang: Line 62:38: **E261** *at least two spaces before inline comment*
- :bangbang: Line 62:39: **E262** *inline comment should start with '# '*
- :bangbang: Line 65:1: **E901** *SyntaxError: non-default argument follows default argument*
- :bangbang: Line 65:101: **E501** *line too long (179 > 100 characters)*
- :bangbang: Line 66:101: **E501** *line too long (183 > 100 characters)*
- :bangbang: Line 68:6: **E111** *indentation is not a multiple of four*
- :bangbang: Line 68:6: **E265** *block comment should start with '# '*
- :bangbang: Line 69:6: **E111** *indentation is not a multiple of four*
- :bangbang: Line 69:6: **E265** *block comment should start with '# '*
- :bangbang: Line 69:101: **E501** *line too long (153 > 100 characters)*
- :bangbang: Line 72:101: **E501** *line too long (101 > 100 characters)*

`utils/tests/test_appliance_cli.py`:
- :bangbang: Line 6:101: **E501** *line too long (101 > 100 characters)*
- :bangbang: Line 9:1: **E302** *expected 2 blank lines, found 1*
- :red_circle: Line 10:50: **F821** *undefined name 'region'*
- :red_circle: Line 10:58: **F821** *undefined name 'internal'*
- :red_circle: Line 10:68: **F821** *undefined name 'dbhostname'*
- :red_circle: Line 10:80: **F821** *undefined name 'username'*
- :red_circle: Line 10:90: **F821** *undefined name 'password'*
- :red_circle: Line 10:100: **F821** *undefined name 'dbname'*
- :bangbang: Line 10:101: **E501** *line too long (141 > 100 characters)*
- :red_circle: Line 10:108: **F821** *undefined name 'key'*
- :red_circle: Line 10:113: **F821** *undefined name 'fetch_key'*
- :red_circle: Line 10:124: **F821** *undefined name 'sshlogin'*
- :red_circle: Line 10:134: **F821** *undefined name 'sshpass'*
- :bangbang: Line 11:101: **E501** *line too long (107 > 100 characters)*
- :bangbang: Line 14:1: **E265** *block comment should start with '# '*
- :bangbang: Line 14:101: **E501** *line too long (105 > 100 characters)*
- :bangbang: Line 16:1: **E265** *block comment should start with '# '*
- :bangbang: Line 18:101: **E501** *line too long (108 > 100 characters)*
- :bangbang: Line 21:1: **E302** *expected 2 blank lines, found 1*
- :bangbang: Line 22:101: **E501** *line too long (102 > 100 characters)*
- :bangbang: Line 23:86: **E261** *at least two spaces before inline comment*
- :bangbang: Line 23:87: **E262** *inline comment should start with '# '*
- :bangbang: Line 23:101: **E501** *line too long (129 > 100 characters)*
- :bangbang: Line 25:101: **E501** *line too long (114 > 100 characters)*
- :bangbang: Line 30:1: **E303** *too many blank lines (3)*

Whoa
Whoa :rage4: ! That is a lot of errors. Please go fix them :point_up: 

Commit inspection report:

**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 428d3095c3483c87ba1996ec59a2624a6c1c521e:

`utils/appliance/__init__.py`:
- :bangbang: Line 66:1: **E901** *SyntaxError: non-default argument follows default argument*

`utils/tests/test_appliance_cli.py`:
- :bangbang: Line 6:101: **E501** *line too long (101 > 100 characters)*
- :bangbang: Line 9:1: **E302** *expected 2 blank lines, found 1*
- :red_circle: Line 10:50: **F821** *undefined name 'region'*
- :red_circle: Line 10:58: **F821** *undefined name 'internal'*
- :red_circle: Line 10:68: **F821** *undefined name 'dbhostname'*
- :red_circle: Line 10:80: **F821** *undefined name 'username'*
- :red_circle: Line 10:90: **F821** *undefined name 'password'*
- :red_circle: Line 10:100: **F821** *undefined name 'dbname'*
- :bangbang: Line 10:101: **E501** *line too long (141 > 100 characters)*
- :red_circle: Line 10:108: **F821** *undefined name 'key'*
- :red_circle: Line 10:113: **F821** *undefined name 'fetch_key'*
- :red_circle: Line 10:124: **F821** *undefined name 'sshlogin'*
- :red_circle: Line 10:134: **F821** *undefined name 'sshpass'*
- :bangbang: Line 11:101: **E501** *line too long (107 > 100 characters)*
- :bangbang: Line 14:1: **E265** *block comment should start with '# '*
- :bangbang: Line 14:101: **E501** *line too long (105 > 100 characters)*
- :bangbang: Line 16:1: **E265** *block comment should start with '# '*
- :bangbang: Line 18:101: **E501** *line too long (108 > 100 characters)*
- :bangbang: Line 21:1: **E302** *expected 2 blank lines, found 1*
- :bangbang: Line 22:101: **E501** *line too long (102 > 100 characters)*
- :bangbang: Line 23:86: **E261** *at least two spaces before inline comment*
- :bangbang: Line 23:87: **E262** *inline comment should start with '# '*
- :bangbang: Line 23:101: **E501** *line too long (129 > 100 characters)*
- :bangbang: Line 25:101: **E501** *line too long (114 > 100 characters)*
- :bangbang: Line 30:1: **E303** *too many blank lines (3)*

Whoa
Whoa :rage4: ! That is a lot of errors. Please go fix them :point_up: 

Commit inspection report:

**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 79328e7a32c0923bcbf63247d116bc05f2d490c2:

`utils/appliance/__init__.py`:
- :bangbang: Line 66:1: **E901** *SyntaxError: non-default argument follows default argument*

`utils/tests/test_appliance_cli.py`:
- :bangbang: Line 7:5: **E122** *continuation line missing indentation or outdented*
- :bangbang: Line 10:1: **E302** *expected 2 blank lines, found 1*
- :red_circle: Line 11:50: **F821** *undefined name 'region'*
- :red_circle: Line 11:58: **F821** *undefined name 'internal'*
- :red_circle: Line 11:68: **F821** *undefined name 'dbhostname'*
- :red_circle: Line 11:80: **F821** *undefined name 'username'*
- :red_circle: Line 12:5: **F821** *undefined name 'password'*
- :red_circle: Line 12:15: **F821** *undefined name 'dbname'*
- :red_circle: Line 12:23: **F821** *undefined name 'key'*
- :red_circle: Line 12:28: **F821** *undefined name 'fetch_key'*
- :red_circle: Line 12:39: **F821** *undefined name 'sshlogin'*
- :red_circle: Line 12:49: **F821** *undefined name 'sshpass'*
- :bangbang: Line 14:5: **E122** *continuation line missing indentation or outdented*
- :bangbang: Line 26:1: **E302** *expected 2 blank lines, found 1*
- :bangbang: Line 28:5: **E122** *continuation line missing indentation or outdented*
- :bangbang: Line 30:5: **E122** *continuation line missing indentation or outdented*
- :bangbang: Line 33:5: **E122** *continuation line missing indentation or outdented*
- :bangbang: Line 38:1: **E303** *too many blank lines (3)*

Whoa
Whoa :rage4: ! That is a lot of errors. Please go fix them :point_up: 

Commit inspection report:

**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit fda2ebf34b8d1290265415ed94c9edeaa829237c:

`utils/appliance/__init__.py`:
- :red_circle: Line 67:24: **F821** *undefined name 'localhost'*
- :red_circle: Line 67:60: **F821** *undefined name 'vmdb_production'*

`utils/tests/test_appliance_cli.py`:
- :bangbang: Line 7:5: **E122** *continuation line missing indentation or outdented*
- :bangbang: Line 10:1: **E302** *expected 2 blank lines, found 1*
- :red_circle: Line 11:50: **F821** *undefined name 'region'*
- :red_circle: Line 11:58: **F821** *undefined name 'internal'*
- :red_circle: Line 11:68: **F821** *undefined name 'dbhostname'*
- :red_circle: Line 11:80: **F821** *undefined name 'username'*
- :red_circle: Line 12:5: **F821** *undefined name 'password'*
- :red_circle: Line 12:15: **F821** *undefined name 'dbname'*
- :red_circle: Line 12:23: **F821** *undefined name 'key'*
- :red_circle: Line 12:28: **F821** *undefined name 'fetch_key'*
- :red_circle: Line 12:39: **F821** *undefined name 'sshlogin'*
- :red_circle: Line 12:49: **F821** *undefined name 'sshpass'*
- :bangbang: Line 14:5: **E122** *continuation line missing indentation or outdented*
- :bangbang: Line 26:1: **E302** *expected 2 blank lines, found 1*
- :bangbang: Line 28:5: **E122** *continuation line missing indentation or outdented*
- :bangbang: Line 30:5: **E122** *continuation line missing indentation or outdented*
- :bangbang: Line 33:5: **E122** *continuation line missing indentation or outdented*
- :bangbang: Line 38:1: **E303** *too many blank lines (3)*

Whoa
Whoa :rage4: ! That is a lot of errors. Please go fix them :point_up: 

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit a2fd3634cb0de2a0acf3b72352a7433c386a8b61:

`utils/tests/test_appliance_cli.py`:
- :bangbang: Line 7:5: **E122** *continuation line missing indentation or outdented*
- :bangbang: Line 10:1: **E302** *expected 2 blank lines, found 1*
- :red_circle: Line 11:50: **F821** *undefined name 'region'*
- :red_circle: Line 11:58: **F821** *undefined name 'internal'*
- :red_circle: Line 11:68: **F821** *undefined name 'dbhostname'*
- :red_circle: Line 11:80: **F821** *undefined name 'username'*
- :red_circle: Line 12:5: **F821** *undefined name 'password'*
- :red_circle: Line 12:15: **F821** *undefined name 'dbname'*
- :red_circle: Line 12:23: **F821** *undefined name 'key'*
- :red_circle: Line 12:28: **F821** *undefined name 'fetch_key'*
- :red_circle: Line 12:39: **F821** *undefined name 'sshlogin'*
- :red_circle: Line 12:49: **F821** *undefined name 'sshpass'*
- :bangbang: Line 14:5: **E122** *continuation line missing indentation or outdented*
- :bangbang: Line 26:1: **E302** *expected 2 blank lines, found 1*
- :bangbang: Line 28:5: **E122** *continuation line missing indentation or outdented*
- :bangbang: Line 30:5: **E122** *continuation line missing indentation or outdented*
- :bangbang: Line 33:5: **E122** *continuation line missing indentation or outdented*
- :bangbang: Line 38:1: **E303** *too many blank lines (3)*

Whoa
Whoa :rage4: ! That is a lot of errors. Please go fix them :point_up: 

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit d27a83923b434538c373913f43df4c120f1d10f5:

`utils/tests/test_appliance_cli.py`:
- :red_circle: Line 12:50: **F821** *undefined name 'region'*
- :red_circle: Line 12:58: **F821** *undefined name 'internal'*
- :red_circle: Line 12:68: **F821** *undefined name 'dbhostname'*
- :red_circle: Line 12:80: **F821** *undefined name 'username'*
- :red_circle: Line 13:5: **F821** *undefined name 'password'*
- :red_circle: Line 13:15: **F821** *undefined name 'dbname'*
- :red_circle: Line 13:23: **F821** *undefined name 'key'*
- :red_circle: Line 13:28: **F821** *undefined name 'fetch_key'*
- :red_circle: Line 13:39: **F821** *undefined name 'sshlogin'*
- :red_circle: Line 13:49: **F821** *undefined name 'sshpass'*
- :bangbang: Line 30:5: **E122** *continuation line missing indentation or outdented*

Whoa
Whoa :rage4: ! That is a lot of errors. Please go fix them :point_up: 

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 4d0da7294baa8fa0c378d151b04fbed2f46c7ccd:

`utils/tests/test_appliance_cli.py`:
- :red_circle: Line 12:50: **F821** *undefined name 'region'*
- :red_circle: Line 12:58: **F821** *undefined name 'internal'*
- :red_circle: Line 12:68: **F821** *undefined name 'dbhostname'*
- :red_circle: Line 12:80: **F821** *undefined name 'username'*
- :red_circle: Line 13:9: **F821** *undefined name 'password'*
- :red_circle: Line 13:19: **F821** *undefined name 'dbname'*
- :red_circle: Line 13:27: **F821** *undefined name 'key'*
- :red_circle: Line 13:32: **F821** *undefined name 'fetch_key'*
- :red_circle: Line 13:43: **F821** *undefined name 'sshlogin'*
- :red_circle: Line 13:53: **F821** *undefined name 'sshpass'*

Whoa
Whoa :rage4: ! That is a lot of errors. Please go fix them :point_up: 

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit bcefb909d81779d02f38f75128d0278b2485cdb5:

`utils/tests/test_appliance_cli.py`:
- :red_circle: Line 12:50: **F821** *undefined name 'region'*
- :red_circle: Line 12:58: **F821** *undefined name 'internal'*
- :red_circle: Line 12:68: **F821** *undefined name 'dbhostname'*
- :red_circle: Line 12:80: **F821** *undefined name 'username'*
- :red_circle: Line 13:9: **F821** *undefined name 'password'*
- :red_circle: Line 13:19: **F821** *undefined name 'dbname'*
- :red_circle: Line 13:27: **F821** *undefined name 'key'*
- :red_circle: Line 13:32: **F821** *undefined name 'fetch_key'*
- :red_circle: Line 13:43: **F821** *undefined name 'sshlogin'*
- :red_circle: Line 13:53: **F821** *undefined name 'sshpass'*
- :red_circle: Line 30:9: **F821** *undefined name 'hostname'*
- :red_circle: Line 30:19: **F821** *undefined name 'domain'*
- :red_circle: Line 30:27: **F821** *undefined name 'realm'*
- :red_circle: Line 30:34: **F821** *undefined name 'username'*
- :red_circle: Line 30:44: **F821** *undefined name 'password'*

Whoa
Whoa :rage4: ! That is a lot of errors. Please go fix them :point_up: 

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 833fedb2b7745b95d51ca7c066e1f3d1b4838a4a:

`utils/appliance/__init__.py`:
- :bangbang: Line 69:8: **E901** *SyntaxError: invalid syntax*
- :bangbang: Line 73:5: **E122** *continuation line missing indentation or outdented*
- :bangbang: Line 106:20: **E251** *unexpected spaces around keyword / parameter equals*
- :bangbang: Line 106:22: **E251** *unexpected spaces around keyword / parameter equals*

`utils/tests/test_appliance_cli.py`:
- :red_circle: Line 28:46: **F821** *undefined name 'db_configuration'*
- :red_circle: Line 29:5: **F841** *local variable 'db_configuration' is assigned to but never used*
- :red_circle: Line 35:5: **F821** *undefined name 'current_appliance'*
- :red_circle: Line 36:9: **F821** *undefined name 'hostname'*
- :red_circle: Line 36:19: **F821** *undefined name 'domain'*
- :red_circle: Line 36:27: **F821** *undefined name 'realm'*
- :red_circle: Line 36:34: **F821** *undefined name 'username'*
- :red_circle: Line 36:44: **F821** *undefined name 'password'*
- :red_circle: Line 37:28: **F821** *undefined name 'current_appliance'*
- :red_circle: Line 40:28: **F821** *undefined name 'current_appliance'*
- :red_circle: Line 46:5: **F821** *undefined name 'current_appliance'*
- :red_circle: Line 47:28: **F821** *undefined name 'current_appliance'*

Whoa
Whoa :rage4: ! That is a lot of errors. Please go fix them :point_up: 

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 0fa33bd80675b2830052b17454eecc8cf1481d6f:

`utils/appliance/__init__.py`:
- :bangbang: Line 69:8: **E901** *SyntaxError: invalid syntax*
- :bangbang: Line 73:5: **E122** *continuation line missing indentation or outdented*
- :bangbang: Line 106:20: **E251** *unexpected spaces around keyword / parameter equals*
- :bangbang: Line 106:22: **E251** *unexpected spaces around keyword / parameter equals*

`utils/tests/test_appliance_cli.py`:
- :bangbang: Line 51:17: **E901** *SyntaxError: invalid syntax*

Please, rectify these issues :smirk: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

File: utils/tests/test_appliance_cli.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        'username': credentials['ipa_server']['username']
+        'password': credentials['ipa_server']['password']
+    }
+
+
+def test_configure_ipa(request, ipa_creds):
+    store.current_appliance.ap_cli.configure_ipa(hostname, domain, realm, username, password)
+    return_code, output = (store.current_appliance.ssh_client.run_command(
+        "systemctl status sssd | grep running"))
+    assert return_code == 0
```
Comment (mfalesni) [inline utils/tests/test_appliance_cli.py]: Two thoughts:
1) How does the non-running service show up? If it says "not running" then you will probably have to grep for that
2) SSH Client now returns a somewhat smart object, therefore you can just do `assert ssh_client.run_command(...)` which automatically assume you want to check for $?=0 :)

File: utils/tests/test_appliance_cli.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            'vmdb_production', fetch_key_ip, app_creds['sshlogin'], app_creds['sshpass'])
+        assert app.is_evm_service_running()
+        app.ipapp.wait_for_web_ui()
+        assert app.is_web_ui_running()
+
+
+@pytest.fixture()
+def ipa_creds():
+    fqdn = cfme_data['ipa']['ipaserver']
+    fqdn.split('.', 1)
```
Comment (psav) [inline utils/tests/test_appliance_cli.py]:
```python
fqdn = cfme_data['ipa']['ipaserver']
fqdn.split('.', 1)
```
There is no assignment here, so the split happens but is lost: so either
```python
fqdn = cfme_data['ipa']['ipaserver'].split('.', 1)
```
or
```python
fqdn = cfme_data['ipa']['ipaserver']
fqdn = fqdn.split('.', 1)
```
I prefer the first

Comment (lcouzens) [inline utils/tests/test_appliance_cli.py]: oh yeh I missed that bit out, but your right the first line you mentioned is a lot nicer.

**Comment by dajoRH** [comment]
Lint report for commit ba85bc7da323a500eb1db9a2c85993e8b5be0dd6:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 3a993f897aeab28ef2bdc5e38219edeb0f256b1a:

`utils/tests/test_appliance_cli.py`:
- :grey_exclamation: Line 37:5: **T001** *print statement found.*

Please, rectify these issues :smirk: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit f687ed10bb6b15c882f76f5420885e948d3a0905:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 0c8f8db7816337967df736f30f065dc7281d0223:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit bee7e46dd0e5ee7a58bb714865b7d21670974901:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 12aa6d9397c092b3e7cb3b4a3c3b6192c27231c9:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit a8e19574091196fc2e97b28cf0afc849e0a71cc5:

Lint OK! :cake: :punch: :cookie: .


**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

File: utils/appliance/__init__.py [hunk 3]
```diff
@@ -58,6 +58,31 @@ class ApplianceException(Exception):
...
+        return self.appliance.ssh_client.run_command(
+            "appliance_console_cli {}".format(ap_cli_command))
+
+    def set_hostname(self, hostname):
+        self.run("-H {}".format(hostname))
+
+    def configure_appliance_internal_fetch_key(self, region, dbhostname,
+            username, password, dbname, fetch_key, sshlogin, sshpass):
+        self.run("-r {} -i -h {} -U {} -p {} -d {} -v -K {} -s {} -a {}".format(
+            region, dbhostname, username, password, dbname, fetch_key, sshlogin, sshpass))
```
Comment (mfalesni) [inline utils/appliance/__init__.py]: Looking at this, we might like to use `utils.quote.quote` to ensure the parameters get passed correctly to the shell. I was thinking about something like this:

```python
from utils.quote import quote as shell_quote
def run(self, command, *args, **kwargs):
    args = [shell_quote(str(arg)) for arg in args]
    kwargs = {k: shell_quote(str(v)) for k, v in kwargs.items()}
    formatted_command = command.format(*args, **kwargs)
    return self.appliance.ssh_client.run_command("appliance_console_cli {}".format(formatted_command))
```

Then the usage would be similar to logger:
```python
self.run('-r {} -i -h {} -U {} -p {} -d {} -v -K {} -s {} -a {}', region, dbhostname, username, password, dbname, fetch_key, sshlogin, sshpass)
```

@psav @RonnyPfannschmidt ?

Comment (psav) [inline utils/appliance/__init__.py]: Could do later I guess, dont' think it needs to hold this up

File: utils/tests/test_appliance_cli.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    apps, pool_id = sp.provision_appliances(
+        count=1, preconfigured=False, version=version, stream=stream)
+
+    yield apps[0]
+
+    sp.destroy_pool(pool_id)
+
+
+@pytest.yield_fixture(scope="module")
+def sprout_provisioned_appliance2():
```
Comment (psav) [inline utils/tests/test_appliance_cli.py]: Let's change this to sprout_fqdn_appliance

Comment (psav) [inline utils/tests/test_appliance_cli.py]: In fact, we don't even really need sprout in there....how about just ```fqdn_appliance``` and ```appliance``` for the one before

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

File: utils/tests/test_appliance_cli.py [hunk 5]
```diff
@@ -0,0 +1,102 @@
+from fixtures.pytest_store import store
+from utils.version import get_stream
+from utils.sprout import SproutClient
```
Comment (psav) [inline utils/tests/test_appliance_cli.py]: Needs replacing with
```from cfme.test_framework.sprout.client import SproutClient```

**Pull request merged**